src/parser.rs

```diff
@@ -27,164 +29,149 @@ impl<'a, T: Read> Parser<T> {
     /// values you can call this multiple times.  If the reader is not yet
     /// ready this will block.
     pub fn parse_value(&mut self) -> RedisResult<Value> {
-        let b = try!(self.read_byte());
-        match b as char {
-            '+' => self.parse_status(),
-            ':' => self.parse_int(),
-            '$' => self.parse_data(),
-            '*' => self.parse_bulk(),
-            '-' => self.parse_error(),
-            _ => fail!((ErrorKind::ResponseError, "Invalid response when parsing value")),
+        let mut buf = vec![];
+        match self.buf_reader.read_to_end(&mut buf) {
```

Comment:
I am working on this now, and am trying to figure out the most efficient way to do so.